mobly/controllers/android_device.py

```diff
@@ -648,13 +649,14 @@ def _terminate_sl4a(self):
         Send terminate signal to sl4a server; stop dispatcher associated with
         the session. Clear corresponding droids and dispatchers from cache.
         """
+        if self.ed:
```

Comment:
Stopping the app after the session teardown is fine.
But I'm not sure why the order is being changed.